src/warp10-datasource.ts

```diff
@@ -296,11 +296,28 @@ export default class Warp10Datasource {
     if (opts.scopedVars) {
       for (let k in opts.scopedVars) {
         let v = opts.scopedVars[k]
-        if (v.selected) {
+        if (v.selected || this.scopedVarIsAll(k)) {
           str += `'${v.value}' '${k}' STORE `
         }
       }
     }
     return str
   }
+
+  /**
+   * Test is a named scoped variable is set to all
+   *
+   * @param name string The name of scoped variable
+   * @return bool If the scoped variable is set to all
+   */
+  private scopedVarIsAll(name): boolean {
+    for (let i = 0; i < this.templateSrv.variables.length; i++) {
+      let v = this.templateSrv.variables[i]
+      if (v.name === name && v.current.value === '$__all') {
```

Comment:
Tips: when you look for only one element in an array matching a condition, you can use [Array.find()](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/find)

Reply:
I pushed a new version with the modifications based on comments. I will look at this one later.